File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/service/RegisterService.java
package net.bestia.zoneserver.service;

import java.util.Objects;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.util.LinkedMultiValueMap;
import org.springframework.util.MultiValueMap;
import org.springframework.web.client.RestTemplate;

import com.fasterxml.jackson.annotation.JsonProperty;

import net.bestia.zoneserver.configuration.StaticConfigService;

/**
 * Performs registrations of the user.
 * 
 * @author Thomas
 *
 */
@Service
public class RegisterService {
	
	private final static String RECAPTCHA_VERIFY_URL = "https://www.google.com/recaptcha/api/siteverify";
	private final StaticConfigService config;
	
	@SuppressWarnings("unused")
	private static class RecaptchaResponse {
		
		public boolean success;
		public String challenge_ts;
		public String hostname;
		
		@JsonProperty("error-codes")
		public String errorCodes;
	}
	
	@Autowired
	public RegisterService(StaticConfigService config) {

		this.config = Objects.requireNonNull(config);
    }
	
	public boolean isHumanUser(String userResponse) {
		
		final RestTemplate rt = new RestTemplate();
		
		// Create a multimap to hold the named parameters		
		final MultiValueMap<String, String> parameters = new LinkedMultiValueMap<String, String>();
		parameters.add("secret", config.getReCaptchaSecretKey());
		parameters.add("response", userResponse);
	
        final RecaptchaResponse response = rt.postForObject(RECAPTCHA_VERIFY_URL, parameters, RecaptchaResponse.class);
        
        return response.success;
    }
}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/client/RegisterService.kt
package net.bestia.zoneserver.client

import com.fasterxml.jackson.annotation.JsonProperty
import net.bestia.zoneserver.configuration.StaticConfig
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.util.LinkedMultiValueMap
import org.springframework.web.client.RestTemplate
import java.util.*

/**
 * Performs registrations of the user and setup of new accounts to the game.
 * This one gets a little tricky since registration might lead to some campaign
 * ids which then get added special items (this will most likly happen by a
 * script).
 *
 * @author Thomas Felix
 */
@Service
class RegisterService @Autowired
constructor(config: StaticConfig) {
  private val config: StaticConfig

  private class RecaptchaResponse {

    var success: Boolean = false
    var challenge_ts: String? = null
    var hostname: String? = null

    @JsonProperty("error-codes")
    var errorCodes: String? = null
  }

  init {

    this.config = Objects.requireNonNull(config)
  }

  fun isHumanUser(userResponse: String): Boolean {

    val rt = RestTemplate()

    // Create a multimap to hold the named parameters
    val parameters = LinkedMultiValueMap<String, String>()
    parameters.add("secret", config.reCaptchaSecretKey)
    parameters.add("response", userResponse)

    val response = rt.postForObject(RECAPTCHA_VERIFY_URL, parameters, RecaptchaResponse::class.java)

    return response.success
  }

  companion object {

    private const val RECAPTCHA_VERIFY_URL = "https://www.google.com/recaptcha/api/siteverify"
  }
}